int16 matrixDim
int16[] dims
float64[] covMat
float64[] meanVec